### ISSUE TITLE
📎 Add `noEmptyTypeParameters` rule

### ISSUE THREAD

**Issue by togami2864** [biomejs/biome#1386]
### Description

originally from: https://github.com/biomejs/biome/issues/1237#issuecomment-1859312147

 (Unfortunately)TypeScript allows Empty type parameters in a type alias declaration or interface declaration. I think we should only handle type alias and interface declarations; anything else will cause parser errors, so we can ignore them.
```js
// input
interface A<> {}
type A<> = {}
```
If it comfirmed, I'll try this
